src/utlls/index.ts

```diff
@@ -31,8 +31,8 @@ export const formatTimeDifference = (ms: number): string => {
 	return `${hours > 0 ? `${hours} h ` : ""}${minutes} m`
 }
 
-const validateScalarArgument = (arg: string, argType: AbiDataType): boolean => {
-	if (argType.endsWith("[]")) {
+const validateScalarArgument = (arg: string | readonly string[], argType: AbiDataType): boolean => {
```

Comment:
This function is not intended to accept an array. Why did you change the input type?